src/services/permissions-checker.js

```diff
@@ -5,18 +5,63 @@ const VError = require('verror');
 
 let permissionsPerRendering = {};
 
-function PermissionsChecker(environmentSecret, renderingId, collectionName, permissionName) {
+function PermissionsChecker(
+  environmentSecret,
+  renderingId,
+  collectionName,
+  permissionName,
+  smartActionId = null,
+  endpoint = null,
+) {
   const EXPIRATION_IN_SECONDS = process.env.FOREST_PERMISSIONS_EXPIRATION_IN_SECONDS || 3600;
 
-  function isAllowed() {
+  function isCollectionAllowed() {
```

Comment:
isAllowed

Reply:
Because it checks smart action and collection. And by the way, is collection allowed is really strange, so isAllowed like before would be nice